### PR TITLE
Server configuration: autogenerated docs improvement

### PR DESCRIPTION
**Pull request by sbesson**
This PR follows up on https://github.com/openmicroscopy/openmicroscopy/pull/3825 which was adding DB properties parsing to the documentation page. Main changes are described as  follows:

- a new configuration section is introduced listing the `Database` server properties
- the `FS` section is renamed as `Binary repository` in the docs to unify with the existing semantics
- a new top-level `Core` section is created with hyperlinks to the fundamental server properties (any better name?)
- in terms of underlying logic, a `Header` object is added allowing to define for each configuration section a display name, a reference (to prevent breakage of `fs_configuration` refs) and a stub for a description (which could be displayed under the title).

To test this PR, check the output of the rST file generated by

```
bin/omero config parse --rst
```

Also check

```
bin/omero config parse --keys
bin/omero config parse --defaults
bin/omero config parse --headers
```
work as expected.

/cc @joshmoore @hflynn @mtbc @ximenesuk

### CODE REVIEW

**Comment by sbesson** [comment]
Deployed at https://www.openmicroscopy.org/site/support/omero5.1-staging/sysadmins/config.html

**Comment by hflynn** [comment]
Staging page looks fine to me. I can't think of a better term than 'core'.

**Comment by joshmoore** [comment]
```
Core
----
  omero.data.dir
  omero.db.host
  omero.db.name
  omero.db.pass
```
doesn't really work for me. Either I think we need some text explaining how we mean "core" or we need something to explain why this section is so short. "Mandatory properties"? "Critical properties (see below)"?

**Comment by sbesson** [comment]
As said, above, adding a short text below a given configuration section should be easy enough with the new `Header` object. Happy to try that if we have a template.

File: etc/omero.properties [hunk 1]
```diff
@@ -67,8 +67,8 @@ omero.db.prepared_statement_cache_size=10
 # you can modify the defaults.
 #
 # Note: some other properties are defined in
-# the file etc/profiles/${omero.db.profile}
-# Especially of importance is omero.db.port
+# the file :file:`etc/profiles/${omero.db.profile}`
```
Comment (manics) [inline etc/omero.properties]: The '`{` `}` are missing from the rendered text.

File: etc/omero.properties [hunk 2]
```diff
@@ -531,7 +531,7 @@ omero.search.ram_buffer_size=64
 # At least one user-owned directory must be included in the path.
 #
 # The template path is created below :property:`omero.managed.dir`,
-# e.g. /OMERO/ManagedRepository/<omero.fs.repo.path>/
+# e.g. :file:`/OMERO/ManagedRepository/${omero.fs.repo.path}/`
```
Comment (manics) [inline etc/omero.properties]: Missing `{` `}`

**Comment by manics** [comment]
Since you've started modifying `etc/omero.properties` I've noticed a few other entries which could do with some sphinx formatting (e.g. filenames, java classnames....)... but I guess you'd rather not do that here.

**Comment by sbesson** [comment]
Feel free to either report them or open a PR against this branch. More commits will be needed to handle this top-level "core" properties section.

File: etc/omero.properties (hunk 1 above)
Comment (sbesson) [inline etc/omero.properties]: Understood, this means Sphinx must escape those then. Should I remove the `file` markup and default to regular verbatim?

Comment (manics) [inline etc/omero.properties]: That's probably best. Someone could go through the whole file and add the relevant markup in a future PR?

Comment (joshmoore) [inline etc/omero.properties]: cF. https://trello.com/c/cATe4yap/306-5-1-omero-properties

**Comment by manics** [comment]
Also `bin/omero config parse --values` doesn't work (did you meant `--headers`?

**Comment by sbesson** [comment]
Yes I meant the different options of `bin/omero config parse` i.e. `--defaults` and `--headers` notably.

**Comment by joshmoore** [comment]
Reads much better to me. Thanks.

**Pull request merged**